### DISCUSSION TITLE
Remove `@remix-run/node` as a direct dependency

### DISCUSSION THREAD

**Discussion by BenJeau** [TanStack/form#1006] [Ideas]
Hi 👋,

Thanks for such a great library! I was looking at the dependency tree of my React vitejs-based application and I saw a lot of dependencies (50) coming from `@tanstack/react-form` related to remix. More than half of my dependencies comes from `@remix-run/node` from `@tanstack/react-form`.

<img width="1464" alt="Screenshot 2024-11-06 at 4 43 08 PM" src="https://github.com/user-attachments/assets/5c63dde2-0398-44f2-954d-0ecfb1b1df20">

Would it be possible to make it optional somehow?

I'm unfamiliar with how npm/node.js works in that regard, but I feel like this dependency might have been missed and shouldn't be included especially if the user of the library isn't using remix.

It seems to only be used here https://github.com/TanStack/form/blob/a4400289b14ad2a9e27a6f9b664891b1604bc163/packages/react-form/src/start/utils.ts

Let me know what you think and if there's a way to achieve this or if its completely unfeasible.

Thanks!

**Comment by mremondi** [comment]
Would also love to see this change. `cookie@0.6.0` has a vulnerability in it (albeit low risk) that I would love to either upgrade or get rid of this dependency altogether 

<img width="1320" alt="Screenshot 2024-11-19 at 11 11 06 AM" src="https://github.com/user-attachments/assets/859443c3-591e-4d64-905e-dd87d2926584">

**Comment by Balastrong** [comment]
That's a valid point, let's check if we still need that package or if we can remove it.

Since the Start adapter is currently broken (breaking changes on Start) it might be the chance to also have a look at this dependency!

See #1023

**Comment by stephen776** [comment]
Hello- I’ll add that the dependency on remix v2.15 packages will break things for anyone trying to use react-router v7 (the upgrade path from remix v2)

**Comment by stephen776** [comment]
Also - can i ask just to clarify… what is the reasoning behind putting the form error state into a cookie in the first place? Just not sure I was following looking at the code for the `start` integration

**Comment by mremondi** [comment]
Hi, any update on this? The majority of my dependabot vulnerabilities come via some subdependency of this library

Most recently due to a vulnerability in undici <= 6.21.0, which is also pulled in via @remix-run/node
